### PR TITLE
feat: reading book

### PR DESCRIPTION
**Pull request by Prerna290**
# PR Template

## Type

- ( ) Fix
- ( ) Refactor
- ( ) Style
- ( ) Docs
- (X) Feat
- ( ) Hotfix

## Bundle Size
7.04 MB

### CODE REVIEW

File: apps/commudle-admin/src/app/feature-modules/public-reading-book/public-reading-book.module.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import { NgModule } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { PublicReadingBookRoutingModule } from './public-reading-book-routing.module';
+import { ReadingBookComponent } from 'apps/commudle-admin/src/app/feature-modules/public-reading-book/components/reading-book/reading-book.component';
+import { NbButtonModule, NbCardModule, NbIconModule, NbListModule, NbSidebarModule } from '@commudle/theme';
+import { ReadingBookIndexComponent } from './components/reading-book-index/reading-book-index.component';
+import { FontAwesomeModule } from '@fortawesome/angular-fontawesome';
+import { SharedComponentsModule } from '../../../../../shared-components/shared-components.module';
+import { SharedPipesModule } from '../../../../../shared-pipes/pipes.module';
```
Comment (ArshdeepGrover) [inline apps/commudle-admin/src/app/feature-modules/public-reading-book/public-reading-book.module.ts]: let's change the path to absolute

File: apps/commudle-admin/src/app/feature-modules/public-reading-book/components/reading-book-index/reading-book-index.component.html [hunk 2]
```diff
@@ -0,0 +1,14 @@
+<h2 class="chapters-heading">Chapters</h2>
+<nb-card>
+  <nb-list>
+    <nb-list-item
+      *ngFor="let chapterIndex of chapterIndexes"
+      [routerLink]="['/developer-community-blueprint', chapterIndex.slug.current]"
+      class="clickable"
+    >
+      <h2 [ngClass]="params === chapterIndex.slug.current ? 'active' : 'title'">
+        {{ chapterIndex.chapter_name }}
```
Comment (ArshdeepGrover) [inline apps/commudle-admin/src/app/feature-modules/public-reading-book/components/reading-book-index/reading-book-index.component.html]: lets put this in anchor tag

File: apps/commudle-admin/src/app/feature-modules/public-reading-book/components/reading-book-index/reading-book-index.component.html [hunk 3]
```diff
@@ -0,0 +1,14 @@
+<h2 class="chapters-heading">Chapters</h2>
+<nb-card>
+  <nb-list>
+    <nb-list-item
+      *ngFor="let chapterIndex of chapterIndexes"
+      [routerLink]="['/developer-community-blueprint', chapterIndex.slug.current]"
+      class="clickable"
+    >
+      <h2 [ngClass]="params === chapterIndex.slug.current ? 'active' : 'title'">
```
Comment (ArshdeepGrover) [inline apps/commudle-admin/src/app/feature-modules/public-reading-book/components/reading-book-index/reading-book-index.component.html]: remove ng class and put routerLinkActive

**Review by ArshdeepGrover — CHANGES_REQUESTED** [review]

**Review by ArshdeepGrover — APPROVED** [review]

File: apps/shared-models/reading_book.model.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+export interface IReadingBookModel {
```
Comment (ArshdeepGrover) [inline apps/shared-models/reading_book.model.ts]: name should be IReadingBook

File: apps/commudle-admin/src/app/feature-modules/public-reading-book/components/reading-book/reading-book.component.ts [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+  }
+
+  ngOnInit(): void {
+    this.authwatchService.currentUser$.subscribe((currentUser) => {
+      this.currentUser = currentUser;
+    });
+    this.getIndex();
+  }
+
+  ngOnDestroy() {
```
Comment (ArshdeepGrover) [inline apps/commudle-admin/src/app/feature-modules/public-reading-book/components/reading-book/reading-book.component.ts:50]: NgOnDestroy should be implement before using

**Review by ArshdeepGrover — CHANGES_REQUESTED** [review]

**Review by ArshdeepGrover — APPROVED** [review]

**Review by arpansac — APPROVED** [review]

**Pull request merged**